iceoryx_hoofs/test/moduletests/test_unix_domain_sockets.cpp

```diff
@@ -69,45 +75,282 @@ class UnixDomainSocket_test : public Test
+TEST_F(UnixDomainSocket_test, UnlinkEmptySocketNameLeadsToInvalidChannelNameError)
 {
-    auto result = UnixDomainSocket::create(
-        goodName, IpcChannelMode::NON_BLOCKING, IpcChannelSide::SERVER, MaxMsgSize, MaxMsgNumber);
-    EXPECT_TRUE(result.has_error());
-    ASSERT_THAT(result.get_error(), Eq(IpcChannelError::INVALID_ARGUMENTS));
+    auto ret = UnixDomainSocket::unlinkIfExists(UnixDomainSocket::NoPathPrefix, "");
+    ASSERT_TRUE(ret.has_error());
+    EXPECT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
+}
+
+TEST_F(UnixDomainSocket_test, UnlinkEmptySocketNameWithPathPrefixLeadsToInvalidChannelNameError)
+{
+    auto ret = UnixDomainSocket::unlinkIfExists("");
+    ASSERT_TRUE(ret.has_error());
+    EXPECT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
+}
+
+TEST_F(UnixDomainSocket_test, UnlinkTooLongSocketNameWithPathPrefixLeadsToInvalidChannelNameError)
+{
+    UnixDomainSocket::UdsName_t longSocketName;
+    for (uint64_t i = 0U; i < UnixDomainSocket::LONGEST_VALID_NAME - strlen(UnixDomainSocket::PATH_PREFIX) + 1; ++i)
+    {
+        longSocketName.append(cxx::TruncateToCapacity, "o");
+    }
+    auto ret = UnixDomainSocket::unlinkIfExists(longSocketName);
+    ASSERT_TRUE(ret.has_error());
+    EXPECT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
+}
+
+TEST_F(UnixDomainSocket_test, UnlinkExistingSocketIsSuccessful)
+{
+    UnixDomainSocket::UdsName_t socketFileName = UnixDomainSocket::PATH_PREFIX;
+    socketFileName.append(cxx::TruncateToCapacity, "iceoryx-hoofs-moduletest.socket");
+    createTestSocket(socketFileName);
+    auto ret = UnixDomainSocket::unlinkIfExists(UnixDomainSocket::NoPathPrefix, socketFileName);
+    EXPECT_FALSE(ret.has_error());
 }
 
-TEST_F(UnixDomainSocket_test, UnlinkNonExistingWithInvalidNameLeadsToError)
+TEST_F(UnixDomainSocket_test, UnlinkExistingSocketWithPathPrefixLeadsIsSuccessful)
 {
-    auto ret = UnixDomainSocket::unlinkIfExists(UnixDomainSocket::NoPathPrefix, invalidName);
-    EXPECT_TRUE(ret.has_error());
-    ASSERT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
+    UnixDomainSocket::UdsName_t socketFileName = "iceoryx-hoofs-moduletest.socket";
+    UnixDomainSocket::UdsName_t socketFileNameWithPrefix = UnixDomainSocket::PATH_PREFIX;
+    socketFileNameWithPrefix.append(cxx::TruncateToCapacity, socketFileName);
+    createTestSocket(socketFileNameWithPrefix);
+    auto ret = UnixDomainSocket::unlinkIfExists(socketFileName);
+    EXPECT_FALSE(ret.has_error());
 }
 
-TEST_F(UnixDomainSocket_test, SendingOnServerLeadsToError)
+void sendOnServerLeadsToError(const sendCall_t& send)
 {
     cxx::string<10> message{"Foo"};
-    Duration duration{1_ms};
-    auto result = server.timedSend(message, duration);
+    auto result = send(message);
     EXPECT_TRUE(result.has_error());
     ASSERT_THAT(result.get_error(), Eq(IpcChannelError::INTERNAL_LOGIC_ERROR));
```

Comment:
This does not work as intended. The `ASSERT_THAT` returns from this function but on the caller side the execution is continued.
This will terminate the test process.
```cpp
void foo() {
    ASSERT_TRUE(false);
}

TEST(Foo, foo) {
    foo();
    auto n = 1/0;
}
```
Currently this function is not used in a harmful way, though

Reply:
You are right but here an `EXPECT_THAT` suffices.